examples/named_dominos.rs

```diff
@@ -0,0 +1,72 @@
+//! An example of using state data to propagate events (See issue-17)
+
+#![deny(missing_docs)]
+
+use smlang::statemachine;
+
+statemachine! {
+    name: Dominos,
```

Comment:
Why have you added so many examples that are copies, but are now named? Wouldn't it be simpler to have one example called `named` or something similar?

Reply:
I wanted a test base to ensure I didn't lose anything in (re)naming things during codegen.